uefi/src/proto/device_path/mod.rs

```diff
@@ -253,6 +253,23 @@ impl PartialEq for DevicePathNode {
     }
 }
 
+impl<'a> TryFrom<&[u8]> for &'a DevicePathNode {
+    type Error = ByteConversionError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if mem::size_of::<DevicePathHeader>() <= bytes.len() {
+            let dp: &DevicePathNode;
+            unsafe {
+                dp = DevicePathNode::from_ffi_ptr(bytes.as_ptr().cast());
```

Comment:
This doesn't look correct, `from_ffi_ptr` doesn't know the length of `bytes`, so it could read past the end.

Reply:
`from_ffi_ptr` only reads `header.length`, which we ensure is within the bytes array on line 260. Then we ensure that `bytes` is large enough to contain the whole node on line 265. However something else might be wrong with this logic, trying to parse through what the miri error is telling me.

Reply:
Even though `from_ffi_ptr` only reads `header.length`, it returns a `DevicePathNode` reference. References in Rust have different validity requirements than raw pointers; the `DevicePathNode` reference must point to a valid object, meaning that the entire (dynamically sized) struct must be valid.

So when the code in  `from_ffi_ptr` does this:
```
        let data_len = usize::from(header.length) - mem::size_of::<DevicePathHeader>();
        &*ptr_meta::from_raw_parts(ptr.cast(), data_len)
```
It would be undefined behavior for `bytes` to be shorter than `data_len`, not just `header.length`.

Reply:
Made #1199 so that I can first read the `DevicePathHeader.length` and ensure that the byte slice is large enough to contain the entire node.